File: CodeForWin/Patterns/Exercise 09/Solution.py
#!/usr/bin/env python3

# Function to print the triangle
def print_triangle(rows):
    rows += 1
    #print row
    for row in range(1, rows):
        #print stars
        for star in range(1, row+1):
            #print first and last stars
            if (row == 1) or (row == rows) or (star == 1) or (star == row+1):
                print("*", end = '')
            else:
                print(" ", end = '')
        print("")

# Main function
def main():
    try:
        rows = int(input("Input row number: "))
        print_triangle(rows)
    except ValueError:
        print("[!] Invalid input")

#Run main function if the script is local
if __name__ == '__main__':
    main()

File: CodeForWin/Patterns/Exercise 09/Solution.c
#include <stdio.h>


int main(){
  //User input
  int rows;
  printf("Input rows number: ");
  scanf("%d", &rows);

  //Print rows
  for (int i = 1; i <= rows; i++){
    //Print stars
    for (int x = 1; x <= i; x++){
      if (i == 1 || i == rows || x == 1 || x == i){
        printf("*");
      } else {
        printf(" ");
      }
    }
    // New line
    printf("\n");
  }

  return 0;
}
